presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/ScalarSubqueryToJoinRewriter.java

```diff
@@ -69,6 +81,58 @@ public ScalarSubqueryToJoinRewriter(FunctionRegistry functionRegistry, SymbolAll
         this.planNodeDecorrelator = new PlanNodeDecorrelator(idAllocator, lookup);
     }
 
+    public Optional<PlanNode> rewriteScalarFilterScan(LateralJoinNode lateralJoinNode, FilterNode filterNode)
```

Comment:
Can you move this to TransformCorrelatedScalarFilterToJoin and make the relevant utility methods public instead?